### PR TITLE
[JENKINS-45661] Defend against and better diagnose corrupted console notes

### PR DESCRIPTION
**Pull request by jglick**
See [JENKINS-45661](https://issues.jenkins-ci.org/browse/JENKINS-45661).

### Proposed changelog entries

* Corrupted console notes could cause an uninformative `NegativeArraySizeException` to be thrown from `ConsoleNote.readFrom` and build log display to be broken.

### CODE REVIEW

**Comment by jglick** [comment]
Build failure is an infrastructure issue.

**Review by jvz — APPROVED** [review]

**Comment by jglick** [comment]
Need to fix up `AnnotatedLargeTextTest`.

File: test/src/test/java/hudson/console/AnnotatedLargeTextTest.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public void oldDeserialization() throws Exception {
         StringWriter w = new StringWriter();
         text.writeHtmlTo(0, w);
         assertEquals("hellothere\n", w.toString());
-        assertThat(logging.getMessages(), hasItem("Failed to resurrect annotation")); // TODO assert that this is IOException: Refusing to deserialize unsigned note from an old log.
+        assertThat(logging.getMessages(), hasItem("Failed to resurrect annotation from \"\\u001B[8mha:AAAAwR+LCAAAAAAAAP9dzLEOwVAUxvHThtiNprYxsGiMQhiwNSIhMR/tSZXr3Lr3oJPwPt7FM5hM3gFh8i3/5Bt+1yeUrYH6ap9Yza1Ys9WKWuMiR05wqWhEgpmyEy306Jxvwb19ccGNoBJjLplmgWq0xgOGCjkNZ2IyTrsRlFayVTs4gVMYqP3pw28/JnznuABF/rYWyIyeJfLQe1vxZiDQ7NnYZLn0UZGRRjA9MiV+0OyFv3+utadQyH8B+aJxVM4AAAA=\\u001B[0mthere\\n\"")); // TODO assert that this is IOException: Refusing to deserialize unsigned note from an old log.
```
Comment (varyvol) [inline test/src/test/java/hudson/console/AnnotatedLargeTextTest.java]: @jglick could you illustrate an example where this kind of log is useful?

Comment (jglick) [reply test/src/test/java/hudson/console/AnnotatedLargeTextTest.java]: See [JIRA](https://issues.jenkins-ci.org/browse/JENKINS-45661?focusedCommentId=363102&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-363102).

**Review by varyvol — APPROVED** [review]

**Pull request merged**